src/gui/plugins/global_illumination_civct/CiVctCascadePrivate.hh

```diff
@@ -0,0 +1,265 @@
+
+    Q_OBJECT
+
+    Q_PROPERTY(
+      float areaHalfSizeX
+      READ AreaHalfSizeX
+      WRITE SetAreaHalfSizeX
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      float areaHalfSizeY
+      READ AreaHalfSizeY
+      WRITE SetAreaHalfSizeY
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      float areaHalfSizeZ
+      READ AreaHalfSizeZ
+      WRITE SetAreaHalfSizeZ
+      NOTIFY SettingsChanged
+    )
+
+    Q_PROPERTY(
+      int resolutionX
+      READ ResolutionX
+      WRITE SetResolutionX
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      int resolutionY
+      READ ResolutionY
+      WRITE SetResolutionY
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      int resolutionZ
+      READ ResolutionZ
+      WRITE SetResolutionZ
+      NOTIFY SettingsChanged
+    )
+
+    Q_PROPERTY(
+      int octantCountX
+      READ OctantCountX
+      WRITE SetOctantCountX
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      int octantCountY
+      READ OctantCountY
+      WRITE SetOctantCountY
+      NOTIFY SettingsChanged
+    )
+    Q_PROPERTY(
+      int octantCountZ
+      READ OctantCountZ
+      WRITE SetOctantCountZ
+      NOTIFY SettingsChanged
+    )
+
+    Q_PROPERTY(
+      float thinWallCounter
+      READ ThinWallCounter
+      WRITE SetThinWallCounter
+      // NOTIFY LightingChanged
+    )
+
+    /// \brief Constructor
+    public: CiVctCascadePrivate(std::mutex &_serviceMutex,
+                                GlobalIlluminationCiVct &_creator,
+                                rendering::CiVctCascadePtr _cascade);
+
+    /// \brief Destructor
+    public: ~CiVctCascadePrivate() override;
+
+    /// \brief Set VCT resolution
+    /// \param[in] _axis Axis (width, height, depth). In range [0; 3)
+    /// \param[in] _res New resolution
+    public: Q_INVOKABLE void UpdateResolution(int _axis, uint32_t _res)
+        EXCLUDES(serviceMutex);
+
+    /// \brief Set VCT octant count
+    /// \param[in] _axis Axis (width, height, depth). In range [0; 3)
+    /// \param[in] _res New octant count
+    public: Q_INVOKABLE void UpdateOctantCount(int _axis, uint32_t _count)
+        EXCLUDES(serviceMutex);
+
+    /// \brief Set the area half size
+    /// \param[in] _axis Axis (width, height, depth). In range [0; 3)
+    /// \param[in] _res New size for that axis
+    public: Q_INVOKABLE void UpdateAreaHalfSize(int _axis, float _halfSize)
+        EXCLUDES(serviceMutex);
+
+    /// \brief Notify various properties may have changed
+    signals: void SettingsChanged();
+
+    /// \brief See rendering::CiVctCascade::SetAreaHalfSize
+    /// \param[in] _enabled See CiVctCascade::SetAreaHalfSize
+    public: Q_INVOKABLE void SetAreaHalfSizeX(const float _x)
+        EXCLUDES(serviceMutex);
+
+    /// \brief See rendering::CiVctCascade::AreaHalfSize
+    /// \return See rendering::CiVctCascade::AreaHalfSize
+    public: Q_INVOKABLE float AreaHalfSizeX() const
+        EXCLUDES(serviceMutex);
```

Comment:
```suggestion
    /// \brief See rendering::CiVctCascade::AreaHalfSizeX
    /// \return See rendering::CiVctCascade::AreaHalfSizeX
    public: Q_INVOKABLE float AreaHalfSizeX() const
        EXCLUDES(serviceMutex);
```

Reply:
Those are not typos. **rendering**::CiVctCascade::AreaHalfSize is the correct function to look for.
I've added extra doc to make this clearer, hopefully